### ISSUE TITLE
CommonJS configuration is not respected and ESM banner is present in final output

### ISSUE THREAD

**Issue by oieduardorabelo** [anomalyco/sst#3878]
We are transition this repo to https://github.com/sst/v2 and are closing this issue for now. If it is still relevant please reopen it there - sorry for the trouble

**Comment by thdxr** [comment]
yep this is a bug i found the issue, will fix

**Comment by thdxr** [comment]
potentially fixed in v3.2.46

**Comment by oieduardorabelo** [comment]
I can confirm SST v3.2.46 outputs the correct CJS.

tested with the following configuration:

```ts
new sst.aws.Function("MyFunction", {
  handler: "hello.handler",
  architecture: "x86_64",
  nodejs: {
    format: "cjs",
  },
});
```

i have a `hello.ts` file:

```ts
exports.handler = async (event, context) => {
  console.log(JSON.stringify(event, null, 2));
  console.log(JSON.stringify(context, null, 2);
};
```

and the output in `.sst/artifacts/MyFunction-src/functions/hello.cjs` is:

```ts
// functions/hello.ts
exports.handler = async (event, context) => {
  console.log(JSON.stringify(event, null, 2));
  console.log(JSON.stringify(context, null, 2));
};
```

thank you @thdxr
